pandas/tests/indexes/test_multi.py

```diff
@@ -1317,6 +1317,18 @@ def test_get_indexer_nearest(self):
         with tm.assertRaises(NotImplementedError):
             midx.get_indexer(['a'], method='pad', tolerance=2)
 
+    def test_hash_collisions(self):
+        # non-smoke test that we don't get hash collisions
+
+        index = MultiIndex.from_product([np.arange(1000), np.arange(1000)],
+                                        names=['one','two'])
+        result = index.get_indexer(index.values)
+        self.assert_numpy_array_equal(result, np.arange(len(index), dtype='int64'))
+
+        for i in [0, 1, len(index)-2, len(index)-1]:
+            result = index.get_loc(index[i])
+            self.assertEqual(result, i)
```

Comment:
I already pushed code to do exactly this. https://github.com/pandas-dev/pandas/pull/15245/commits/99f7cc1eb6e113b9b38188e6099d01aa8ad3e683

I do a run-time check if there is a collision by comparison. It actually turned up a bug (separate)